### PR TITLE
chore: update `with-tailwindcss` example to TypeScript

### PR DESCRIPTION
**Pull request by saulloalmeida**
## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `yarn lint`
- [ ] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/with-tailwindcss/package.json [hunk 1]
```diff
@@ -3,21 +3,23 @@
...
     "build": "next build",
-    "start": "next start"
+    "start": "next start",
+    "lint": "next lint"
   },
   "dependencies": {
-    "next": "latest",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "next": "12.1.6",
```
Comment (balazsorban44) [inline examples/with-tailwindcss/package.json:10]: This should not be pinned. Let's keep `latest`:

```suggestion
    "next": "latest",
```

File: examples/with-tailwindcss/package.json [hunk 2]
```diff
@@ -3,21 +3,23 @@
...
-    "prettier": "^2.5.1",
-    "prettier-plugin-tailwindcss": "^0.1.1",
-    "tailwindcss": "^3.0.7",
-    "typescript": "4.5.4"
+    "@types/node": "17.0.35",
+    "@types/react": "18.0.9",
+    "@types/react-dom": "18.0.5",
+    "autoprefixer": "^10.4.7",
+    "eslint": "8.16.0",
+    "eslint-config-next": "12.1.6",
```
Comment (balazsorban44) [inline examples/with-tailwindcss/package.json:20]:
```suggestion
```

ESLint should not be added, as per our recently updated contributing guidelines for examples: https://github.com/vercel/next.js/pull/37193

**Review by balazsorban44 — DISMISSED** [review]
Thanks, I added some comments, please review

File: examples/with-tailwindcss/package.json [hunk 3]
```diff
@@ -3,21 +3,23 @@
...
+    "postcss": "^8.4.14",
+    "tailwindcss": "^3.0.24",
+    "typescript": "4.7.2"
```
[the remark this answers is not in the document]
Comment (ijjk) [reply examples/with-tailwindcss/package.json:20]: Weirdly seems there was a `tsconfig.json` [already](https://github.com/vercel/next.js/blob/canary/examples/with-tailwindcss/tsconfig.json) but good catch checking this!

**Review by ijjk — APPROVED** [review]
Thanks!

**Pull request merged**